pkg/server/server_controller_orchestration.go

```diff
@@ -170,10 +177,16 @@ func (c *serverController) scanTenantsForRunnableServices(
 	return nil
 }
 
+// createServerEntryLocked starts an orchestration task for the given
+// tenant.
 func (c *serverController) createServerEntryLocked(
 	ctx context.Context, tenantName roachpb.TenantName,
 ) (*serverEntry, error) {
-	entry, err := c.startControlledServer(ctx, tenantName)
+	// We need to set singleShot to false because it is
+	// non-deterministic whether we use this code path synchronously or
+	// whether the controller has started the service asynchronously as
+	// part of scanTenantsForRunnableServices().
```

Comment:
There aren't many call-sites, so I suppose if we wanted to use single-shot in the testserver user-case, we could expose it as an argument to this function.

But, fine as is, no reason to change.